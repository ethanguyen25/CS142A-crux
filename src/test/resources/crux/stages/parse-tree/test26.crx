// test loop

int a;

void main() {
  a = 9;
  loop {
    if (a == 0) {
      break;
    }
    printInt(a);
    println();
    a = a-1;
  }
  printInt(0-5);
  printInt(a);
}
