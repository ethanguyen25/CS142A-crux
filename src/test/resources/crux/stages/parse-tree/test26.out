program
  declarationList
    declaration
      variableDeclaration
        type
    declaration
      functionDefinition
        type
        parameterList
        statementBlock
          statementList
            statement
              assignmentStatement
                designator
                expression0
                  expression1
                    expression2
                      expression3
                        literal
            statement
              loopStatement
                statementBlock
                  statementList
                    statement
                      ifStatement
                        expression0
                          expression1
                            expression2
                              expression3
                                expression0
                                  expression1
                                    expression2
                                      expression3
                                        designator
                                  op0
                                  expression1
                                    expression2
                                      expression3
                                        literal
                        statementBlock
                          statementList
                            statement
                              breakStatement
                    statement
                      callStatement
                        callExpression
                          expressionList
                            expression0
                              expression1
                                expression2
                                  expression3
                                    designator
                    statement
                      callStatement
                        callExpression
                          expressionList
                    statement
                      assignmentStatement
                        designator
                        expression0
                          expression1
                            expression1
                              expression2
                                expression3
                                  designator
                            op1
                            expression2
                              expression3
                                literal
            statement
              callStatement
                callExpression
                  expressionList
                    expression0
                      expression1
                        expression1
                          expression2
                            expression3
                              literal
                        op1
                        expression2
                          expression3
                            literal
            statement
              callStatement
                callExpression
                  expressionList
                    expression0
                      expression1
                        expression2
                          expression3
                            designator
